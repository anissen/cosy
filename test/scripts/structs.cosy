{
    struct S { mut x = 4 }
    print S
    print S { x = 6 }
    print S
    mut s = S {}
    print s
    s.x = 1
    print s
}

struct Point {
	mut x Num = 2
    var y = 4
    var a Array Num
}

{
    mut point = Point { x = 5, y = 13, a = [] } // ok
    print point
}

{
    struct Test {
        var a Point
        var b Str
    }

    print Test { a = Point { x = 666, y = 777, a = [] }, b = 'hej' }
}

{
    struct Test2 {
        var a Array Num
        var p Point = Point { x = 4, y = 12, a = [] }
    }

    var test = Test2 { a = [3, 4, 5] }
    print test
}

{
    fn struct_as_arg(s) {
        print s
    }
    struct_as_arg(Point { x = 3, y = 4, a = [] })
}

{
    struct Entity {
        var x Num
        var y Num
        var text = 'default value'
        mut arr Array Num = []
    }

    // instantiate struct
    mut entity = Entity { x = 3, y = 5 }
    entity.arr = [5, 6, 7]
    print entity
}

{
	struct S {
		mut x Num
	}
	mut s = S { x = 42}
	print s
	s.x = 45
	print s
	s.x += 45
	print s
	s.x -= 25
	print s
}

{
    struct Obj {
        mut x Num
    }
    mut arr = [Obj { x = 1}]
    print arr
    mut element = arr[0]
    element.x = 45
    print arr
}