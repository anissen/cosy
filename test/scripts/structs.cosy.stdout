S { x = 4 }
S { x = 6 }
S { x = 4 }
S { x = 4 }
S { x = 1 }
Point { a = [], x = 5, y = 13 }
Test { a = Point { a = [], x = 666, y = 777 }, b = "hej" }
Test2 { a = [3,4,5], p = Point { a = [], x = 4, y = 12 } }
Point { a = [], x = 3, y = 4 }
Entity { arr = [5,6,7], text = "default value", x = 3, y = 5 }
S { x = 42 }
S { x = 45 }
S { x = 90 }
S { x = 65 }
